fix Issue 4149 - refs displayed as pointers in gdb

MartinNowak opened the pull request:
- use TYnref for reference variables
- emit dwarf debug info for references
- handle TY*ref in backend
- add TYFLref to tyscalar

[Issue 4149 – refs displayed as pointers in gdb](https://issues.dlang.org/show_bug.cgi?id=4149)

WalterBright commented on src/backend/cgcv.c:
```diff
@@ -2162,6 +2162,10 @@ unsigned cv4_typidx(type *t)
         case TYref:
+        case TYnref:
+#if TARGET_SEGMENTED
+        case TYfref:
```
You can remove all the `TYfref`s. They are never going to happen, as D is never going to support 16 bit segmented memory models.

MartinNowak replied:
done

MartinNowak committed:
fix Issue 4149 - refs displayed as pointers in gdb

- use TYnref for reference variables
- emit dwarf debug info for references
- handle TY*ref in backend
- add TYFLref to tyscalar
